### ISSUE TITLE
TPU support update

### ISSUE THREAD

**Issue by antoinegargot** [CyberZHG/keras-bert#64]
Hi,

First, thank you for all the work you already did to make Bert works with Keras.

I was wondering if updating the current version to run on TPU was on the roadmap? (the old version https://github.com/HighCWu/keras-bert-tpu doesn't implement all the modification that you recently did) 

Thank you!

**Comment by CyberZHG** [comment]
I added an example of how to run it on TPU:

https://colab.research.google.com/github/CyberZHG/keras-bert/blob/master/demo/load_model/keras_bert_load_and_extract_tpu.ipynb

However, the codes are too platform-specific and it's hard to write unit tests, therefore I don't want to add it to released packages.

**Comment by antoinegargot** [comment]
Thank you for the help! 

Makes complete sense that you don't want to release a TPU-ready code for now. 
Do you know if the AdamWarmup Optimizer is compatible with tf.keras for TPU conversion?

**Comment by CyberZHG** [comment]
The optimizer can be used on TPU but I don't know whether the configs are synced correctly. See:

https://colab.research.google.com/github/CyberZHG/keras-bert/blob/master/demo/load_model/keras_bert_load_and_fit_tpu.ipynb

**Comment by antoinegargot** [comment]
Thanks! That was pretty quick, I will give it a try. As a note for others using this notebook as a reference, I will add that the param decay steps and warmup step should depend on the dataset you are training the model on, I set them as follow:
* decay_steps = int(number_of_epochs*len(training_set)/batch_size)
* warmup_steps = int(0.1*decay_steps)
